.github/workflows/docker-tests.yml

```diff
@@ -0,0 +1,164 @@
+    -pl minifi/minifi-assembly
+    -pl minifi/minifi-c2/minifi-c2-assembly
+
+  MAVEN_CLEAN_ARGUMENTS: >-
+    -T 2.0C
+    clean
+  MAVEN_CLEAN_PROJECTS: >-
+    -pl -nifi-assembly
+    -pl -nifi-registry/nifi-registry-assembly
+    -pl -nifi-toolkit/nifi-toolkit-assembly
+    -pl -minifi/minifi-assembly
+    -pl -minifi/minifi-c2/minifi-c2-assembly
+
+# build and test Docker Images
+  MAVEN_DOCKER_ARGUMENTS: >-
+    install
+    -P docker
+  MAVEN_DOCKER_PROJECTS: >-
+    -pl minifi/minifi-docker
+    -pl minifi/minifi-c2/minifi-c2-docker
+    -pl nifi-docker/dockermaven
+    -pl nifi-registry/nifi-registry-docker-maven/dockermaven
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    strategy:
+      fail-fast: false
+      matrix:
+        # TODO: macos-latest doesn't come with Docker and Colima seems to have problems with the nifi-docker integration_test scripts - see NIFI-12177/NIFI-12191
+        os: [ ubuntu-latest ]
+        version: [ 21 ]
+    timeout-minutes: 120
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} Java ${{ matrix.version }}
+    env:
+      # default to Amazon Corretto JDK
+      JAVA_DISTRIBUTION: corretto
+    steps:
+      # Zulu JDK appears to be more reliable for MacOS
+      - name: Use Java distribution Zulu
+        if: ${{ runner.os == 'macos' }}
+        run: echo "JAVA_DISTRIBUTION=zulu" >> "$GITHUB_ENV"
+      - name: Checkout Code
+        uses: actions/checkout@v4
+      - name: Set up Java ${{ env.JAVA_DISTRIBUTION }} ${{ matrix.version }}
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ matrix.version }}
+          cache: 'maven'
+
+      - name: Set up Docker
+        if: ${{ runner.os == 'macos' }}
+        run: |-
+          /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+          brew install docker
+          colima start
+
+          if [ ! -e /var/run/docker.sock ]; then sudo ln -s ~/.colima/docker.sock /var/run/docker.sock; fi
+      - name: Confirm Docker
+        run: id && docker info
+
+      - name: Build Assemblies
+        env:
+          MAVEN_OPTS: >-
+            ${{ env.DEFAULT_MAVEN_OPTS }}
+        run: >
+          ${{ env.MAVEN_COMMAND }}
+          ${{ env.MAVEN_BUILD_ARGUMENTS }}
+          ${{ env.MAVEN_BUILD_PROJECTS }}
+      - name: Clean Modules
+        env:
+          MAVEN_OPTS: >-
+            ${{ env.DEFAULT_MAVEN_OPTS }}
+        # clean non-assembly modules before running Docker builds and delete unpacked binaries, otherwise we'll run out of disk space
+        run: >
+          ${{ env.MAVEN_COMMAND }}
+          ${{ env.MAVEN_CLEAN_ARGUMENTS }}
+          ${{ env.MAVEN_CLEAN_PROJECTS }}
+          && rm -rf nifi-assembly/target/nifi-*-bin/ nifi-registry/nifi-registry-assembly/target/nifi-registry-*-bin/
```

Comment:
Instead of manually removing the binaries, building the registry-assembly binary should be disabled for this particular workflow.

Reply:
We need the binaries, but only the Zip file versions, these `rm` commands remove the unzipped directories from `target` (which exist prior to the `.zip` being created by the assembly plugin)

This *may not* be required for the space saving any longer, but I had a lot of issues with disk space during development, so deleting unnecessary files between steps seemed most sensible

Reply:
Can you revisit this based on the latest version and see if it is still necessary? It would be best to avoid including the removal if it is not needed, or call the appropriate build settings to skip Zip binary generation.